src/main/java/de/zalando/aruha/nakadi/config/SecurityConfiguration.java

```diff
@@ -1,46 +1,75 @@
 package de.zalando.aruha.nakadi.config;
 
-import java.text.MessageFormat;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-
-import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
 import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
 import org.springframework.security.oauth2.config.annotation.web.configurers.ResourceServerSecurityConfigurer;
 import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
 
-import org.zalando.stups.oauth2.spring.server.TokenInfoResourceServerTokenServices;
+import java.text.MessageFormat;
+
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpMethod.PUT;
 
 @EnableResourceServer
 @Configuration
 public class SecurityConfiguration extends ResourceServerConfigurerAdapter {
 
     private static final Logger LOG = LoggerFactory.getLogger(SecurityConfiguration.class);
-    public static final String UID = "uid";
 
-    @Value("${nakadi.oauth2.tokenInfoUri}")
-    private String tokenInfoUri;
+    @Autowired
+    private SecuritySettings settings;
+
+    @Autowired
+    private ResourceServerTokenServices tokenServices;
 
-    @Value("${nakadi.oauth2.clientId}")
-    private String clientId;
+    @Value("${nakadi.oauth2.scopes.uid}")
+    private String uidScope;
 
-    @Value("${nakadi.oauth2.enabled:true}")
-    private boolean oauthEnabled;
+    @Value("${nakadi.oauth2.scopes.nakadiRead}")
+    private String nakadiReadScope;
+
+    @Value("${nakadi.oauth2.scopes.eventTypeWrite}")
+    private String eventTypeWriteScope;
+
+    @Value("${nakadi.oauth2.scopes.eventStreamRead}")
+    private String eventStreamReadScope;
+
+    @Value("${nakadi.oauth2.scopes.eventStreamWrite}")
+    private String eventStreamWriteScope;
 
     @Override
     public void configure(final HttpSecurity http) throws Exception {
-        LOG.info("Security: OAuth2 is " + (oauthEnabled ? "enabled" : "disabled"));
-        if (oauthEnabled) {
-            http.authorizeRequests().antMatchers("/health").permitAll().anyRequest().access(hasScope(UID));
-        } else {
-            http.authorizeRequests().anyRequest().permitAll();
+        LOG.info("Authentication mode: " + settings.getAuthMode());
+
+        if (settings.getAuthMode() == SecuritySettings.AuthMode.FULL) {
+            http.authorizeRequests()
+                    .antMatchers("/health").permitAll()
+                    .antMatchers(GET, "/metrics").access(hasScope(uidScope))
```

Comment:
I don't know if we should have the specific scope for accessing /metrics endpoint. I assumed that we don't.